docs/source/config.rst

```diff
@@ -0,0 +1,64 @@
+Configuring Jupyter applications
+================================
+
+The Jupyter applications have a common config system.
+
+.. _config_dir:
+
+The Jupyter config directory
+----------------------------
+
+Config files are stored in :file:`~/.jupyter` by default.
+
+.. envvar:: JUPYTER_CONFIG_DIR
+
+   Set this environment variable to use another location for Jupyter config files.
+
+Python config files
+-------------------
+
+To create the blank config files, run::
+
+    jupyter {application} --generate-config
+
+The file will be named :file:`jupyter_{application}.py`
+
+Inside the file, you can configure class attributes like this::
+
+    c.NotebookApp.port = 8754
+
+Be careful with spelling--incorrect names will simply be ignored, with
+no error.
+
+To add to a collection which may have already been defined elsewhere,
+you can use methods like those found on lists, dicts and sets: append,
+extend, :meth:`~traitlets.config.LazyConfigValue.prepend` (like
+extend, but at the front), add and update (which works both for dicts
+and sets)::
+
+    c.TemplateExporter.template_path.append('./templates')
+
+
+Command line arguments
+----------------------
+
+Every configurable value can be set from the command line, using this
+syntax::
+
+    jupyter {application} --ClassName.attribute=value
```

Comment:
`ClassName.attribute` seem to be an implementation detail ? I would use the same example as above with `--NotebookApp.port=8754` and a "just replace `c.` by `==`"